src/tribler/gui/port_checker.py

```diff
@@ -0,0 +1,56 @@
+import socket
+
+import psutil
+from PyQt5.QtCore import QObject, QTimer
+
+from tribler.gui.utilities import connect
+
+CHECK_INTERVAL_MS = 1000
+MAX_PORTS_TO_CHECK = 10
+
+
+class PortChecker(QObject):
+
+    def __init__(self, base_port, callback):
+        QObject.__init__(self, None)
+        self.base_port = base_port
+        self.callback = callback
+
+        self.process_id = None
+        self.checker_timer = None
+
+    def setup_with_pid(self, pid):
+        self.process_id = pid
+        self.checker_timer = QTimer()
+        self.checker_timer.setSingleShot(True)
+        connect(self.checker_timer.timeout, self.on_timeout)
+        self.checker_timer.start(CHECK_INTERVAL_MS)
+
+    def on_timeout(self):
+        detected_port = self.detected_port()
+        if detected_port:
+            self.checker_timer.stop()
+            self.callback(detected_port)
+        elif self.checker_timer:
```

Comment:
There were some print statements in the previous version of `PortChecker`. With them, it was possible to see that `PortChecker` never stops on Windows. First, it detects an incorrect `processId` (as I described above), and then it continues to check open ports again and again even after `Tribler` has already started and GUI has connected to Core.

So, I have two suggestions here:
1. It is useful to have some logging output from `PortChecker` so we can detect that it runs longer than expected in case the process ID was detected incorrectly.
2. After the GUI process is able to connect to Core successfully, it should explicitly stop `PortChecker`.